pkg/server/server_test.go

```
@@ -1 +1,386 @@
+		t.Fatalf("expected nil, got %+v", got)
+	}
+
+	// Anonymous fallback rule reachable for public/* with the "_anonymous" key.
+	got = p.Authorize(AnonymousUser, AccessScope{{Type: RepositoryAccessType, Name: "public/x"}})
+	if len(got) != 1 || got[0].Actions[0] != PullAction {
+		t.Fatalf("anonymous authorize: %+v", got)
+	}
+}
+
+func TestDecodeScope_FromQuery(t *testing.T) {
+	req := httptest.NewRequest("GET", "http://x/auth/token?service=svc&scope=repository:foo/bar:pull,push", nil)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)